stackslib/src/net/stackerdb/config.rs

```diff
@@ -0,0 +1,451 @@
+            );
+            return Err(net_error::InvalidStackerDBContract(contract_id.clone()));
+        }
+
+        let max_neighbors = config_tuple
+            .get("max-neighbors")
+            .expect("FATAL: missing 'max-neighbors'")
+            .clone()
+            .expect_u128();
+        if max_neighbors > usize::MAX as u128 {
+            debug!(
+                "Contract {} stipulates a maximum number of neighbors beyond usize::MAX",
+                contract_id
+            );
+            return Err(net_error::InvalidStackerDBContract(contract_id.clone()));
+        }
+
+        let hint_replicas_list = config_tuple
+            .get("hint-replicas")
+            .expect("FATAL: missing 'hint-replicas'")
+            .clone()
+            .expect_list();
+        let mut hint_replicas = vec![];
+        for hint_replica_value in hint_replicas_list.into_iter() {
+            let hint_replica_data = hint_replica_value.expect_tuple();
+
+            let addr_byte_list = hint_replica_data
+                .get("addr")
+                .expect("FATAL: missing 'addr'")
+                .clone()
+                .expect_list();
+            let port = hint_replica_data
+                .get("port")
+                .expect("FATAL: missing 'port'")
+                .clone()
+                .expect_u128();
+            let pubkey_hash_bytes = hint_replica_data
+                .get("public-key-hash")
+                .expect("FATAL: missing 'public-key-hash")
+                .clone()
+                .expect_buff_padded(20, 0);
+
+            let mut addr_bytes = vec![];
+            for byte_val in addr_byte_list.into_iter() {
+                let byte = byte_val.expect_u128();
+                if byte > (u8::MAX as u128) {
+                    debug!(
+                        "Contract {} stipulates an addr byte above u8::MAX",
+                        contract_id
+                    );
+                    return Err(net_error::InvalidStackerDBContract(contract_id.clone()));
+                }
+                addr_bytes.push(byte as u8);
+            }
+            if addr_bytes.len() != 16 {
+                debug!(
+                    "Contract {} did not stipulate a full 16-octet IP address",
+                    contract_id
+                );
+                return Err(net_error::InvalidStackerDBContract(contract_id.clone()));
+            }
+
+            if port < 1024 || port > ((u16::MAX - 1) as u128) {
+                debug!(
+                    "Contract {} stipulates a port lower than 1024 or above u16::MAX - 1",
+                    contract_id
+                );
+                return Err(net_error::InvalidStackerDBContract(contract_id.clone()));
+            }
+
+            let mut pubkey_hash_slice = [0u8; 20];
+            pubkey_hash_slice.copy_from_slice(&pubkey_hash_bytes[0..20]);
+
+            let peer_addr = PeerAddress::from_slice(&addr_bytes).expect("FATAL: not 16 bytes");
```

Comment:
Not an issue with this PR, but it surprised me that PeerAddress::from_slice returned an optional rather than a Result. Is there a particular reason? Would be nice to have consistency through the code base for result vs optional for try style constructors.